build/__init__.py

```diff
@@ -70,11 +70,12 @@ def check_version(requirement_string, extra=''):  # type: (str, str) -> bool
 
 
 class ProjectBuilder(object):
-    def __init__(self, srcdir='.'):  # type: (str) -> None
+    def __init__(self, srcdir='.', config_settings={}):  # type: (str, Dict[str, Union[str, List[str]]]) -> None
         '''
         :param srcdir: Source directory
         '''
         self.srcdir = srcdir
+        self.config_settings = config_settings
```

Comment:
Is it ok to share the dict here?  (compared to setting a new empty dict if received param is None)

Reply:
Yeah this definitely, you should never use mutable objects as default.